### ISSUE TITLE
Technical Fork: Continuous Metric Release Candidate

### ISSUE THREAD

**Issue by vinodkahuja** [chaoss/wg-common#94]
This issue is created to collect comments about the rolling release of "Technical Fork"

This thread is for comments about the proposed metric Technical Fork.

**Comment by klumb** [comment]
The metric can be found here: https://github.com/chaoss/wg-common/blob/master/focus-areas/what/technical-fork.md

See all release candidates: https://chaoss.community/metrics/

**Comment by rpaik** [comment]
Any concern that this may be viewed as too GitHub (or even GitLab) centric? I assume this is meant to track how many people fork/clone projects and then contribute to the main project, but this may not translate to projects that are not using GitHub/GitLab. 

Here's a good article: https://opensource.com/article/17/12/fork-clone-difference

**Comment by germonprez** [comment]
Nice point. Clone might be the term here... the article was great.

**Comment by GeorgLink** [comment]
We struggled with the terminology for this metric. We settled on "technical fork" to distinguish between "hostile fork" or "community fork". 

During the call today, the term "clone" received much support. My concern with "clone" is that in GitHub it refers to making a local copy only.

**Comment by klumb** [comment]
At the basic level, this metric describes how many copies of a project are made. I don't think it cares about the intent or where the copy is made. 

One of the problems with the term fork is that _intentions_ are often included in the definition. 

For that reason, I do prefer clone or copy.

**Comment by vinodkahuja** [comment]
After reading the article @rpaik pointed I would also vote for the clone. The term fork may include the platform bias.

**Comment by GeorgLink** [comment]
I sense a wide support for changing the metric name and description,
replacing `fork` with `clone`.

Creating a pull request for this change could be volunteer opportunity.

**Comment by rpaik** [comment]
I probably opened a can of worms, but I feel like there's confusion over clone vs. fork (or even branch) in general. Maybe "Technical copy" or even "Project copy" could also be options?

**Comment by germonprez** [comment]
I don't have a problem with it @rpaik :) I think the discussion is great!

**Comment by conferencepapers** [comment]
This metric is a bit tricky because it discusses the intercession between a fork and a 'clone'. In particular, we can't just result in a clone because we can't measure a bare clone except when activity happens in the local clone that reflects back to the main project, in this case, we are talking about branching. On the other hand, we can measure forks directly but, forks usually take us away from the parent project.

So, if we want to measure the contribution activities around a fork/clone of a project, we might consider the number of branches that exist within the project codebase.

**Comment by foundjem** [comment]
This metric is a bit tricky because it discusses the intercession between a fork and a 'clone'. In particular, we can't just result in a clone because we can't measure a bare clone except when activity happens in the local clone that reflects back to the main project, in this case, we are talking about branching. On the other hand, we can measure forks directly but, forks usually take us away from the parent project.
So, if we want to measure the contribution activities around a fork/clone of a project, we might consider the number of branches that exist within the project codebase.

**Comment by germonprez** [comment]
Yes, this metric is tricky but based on the March 4th Common meeting we decided to keep it as is. However, this includes some work that will need to done in the future. This work will include new metrics for CLONES and DOWNLOADS and will likely include new metrics regarding the nature of the technical fork (hostile or developmental) to be considered in other WGs.

**Comment by klumb** [comment]
Closing issue for release

**Comment by vinodkahuja** [comment]
Some contents of this metric are changed when we were developing the clone metric and made the language in syn to clone metric.

This thread is now for collecting comments about the changes to the released technical fork metric. 

Metric is available at
https://github.com/chaoss/wg-common/blob/master/focus-areas/contributions/technical-fork.md

See all release candidates of metrics are at:
https://chaoss.community/metrics/
